server/public/model/config_test.go

```diff
@@ -1601,6 +1610,56 @@ func TestPluginSettingsSanitize(t *testing.T) {
 	}
 }
 
+func TestSanitizeDataSource(t *testing.T) {
+	t.Run(DatabaseDriverPostgres, func(t *testing.T) {
+		testCases := []struct {
+			Original  string
+			Sanitized string
+		}{
+			{
+				"",
+				"//****:****@",
```

Comment:
I understand this was ported as is, but this is weird. In my mind, an empty input should either produce an empty output or fail altogether.